src/AutoRest.CSharp.V3/Generation/Writers/RestClientWriter.cs

```diff
@@ -191,6 +196,40 @@ private void WriteRequestCreation(CodeWriter writer, RestClientMethod clientMeth
                             writer.Line($");");
                         }
                         break;
+                    case MultipartRequestBody multipartRequestBody:
+                        var multipartContent = new CodeWriterDeclaration("content");
+                        using (WriteValueNullCheck(writer, multipartRequestBody.Value))
+                        {
+                            writer.Line($"var {multipartContent:D} = new MultipartFormDataContent();");
+                            writer.Line($"{multipartContent}.ApplyToRequest({request});");
+                            if (multipartRequestBody.Value.Type.Name == "IEnumerable")
+                            {
+                                var collectionItemVariable = new CodeWriterDeclaration("value");
+                                using (writer.Scope($"foreach (var {collectionItemVariable:D} in {multipartRequestBody.Value.Reference.Name})"))
+                                {
+                                    writer.Append($"{multipartContent}.Add({typeof(RequestContent)}.Create({collectionItemVariable}");
+                                    writer.Append($"), \"{multipartRequestBody.Value.Reference.Name}\"");
+                                    writer.Line($", null);");
+                                }
+                            }
+                            else
+                            {
+                                writer.Append($"{multipartContent}.Add({typeof(RequestContent)}.Create(");
+                                WriteConstantOrParameter(writer, multipartRequestBody.Value);
+                                writer.Append($"), \"{multipartRequestBody.Value.Reference.Name}\"");
+                                writer.Line($", null);");
+                            }
+                        }
+
+                            if (multipartRequestBody.Name != null)
+                            {
+                                writer.Append($"{multipartContent}.Add(new {typeof(StringRequestContent)}(");
+                                writer.AppendRaw($"{multipartRequestBody.Name?.Reference.Name}");
+                                writer.Append($"), \"{multipartRequestBody.Name?.Reference.Name}\"");
+                                writer.Line($", null);");
+                            }
+                            writer.Line($"{request}.Content = {multipartContent};");
```

Comment:
Doesn't `ApplyToRequest` already do this?

Reply:
Nope. Here is a PR for this: https://github.com/Azure/azure-sdk-for-net/pull/14925